metatensor-torch/src/block.cpp

```diff
@@ -158,6 +158,13 @@ torch::Tensor TensorBlockHolder::values() const {
     return wrapper->tensor();
 }
 
+void TensorBlockHolder::set_values(const torch::Tensor& new_values) {
+        throw std::runtime_error(
+            "Direct assignment to `values` is not allowed. "
+            "Please use block.values().copy_(new_values) instead."
```

Comment:
why not `block.values[:] = new_values`? Since this error will be seen from Python, it might make sense to use the Python syntax as well!